response_operations_ui/views/collection_exercise.py

```diff
@@ -162,6 +163,16 @@ def _validate_collection_instrument():
                 "header": "Error: wrong file type for Collection instrument",
                 "message": "Please use XLSX file only"
             }
+        else:
+            # file name format is surveyId_period_formType
+            form_type = _get_form_type(file.filename) if file.filename.count('_') == 2 else None
+            if not form_type.isdigit() and len(form_type) != 4:
```

Comment:
`.isdigit()` will surely raise an AttributeError if form_type evaluates to None in the line above.

Reply:
Likewise, `len(None)` will raise a TypeError.

Reply:
Nice one, found a missing test case too 👍